runelite-client/src/main/java/net/runelite/client/plugins/pvptools/PvpToolsPlugin.java

```diff
@@ -422,6 +425,14 @@ private void onPlayerDespawned(PlayerDespawned event)
 		}
 	}
 
+	private void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (config.hideCastNpcs() && event.getType() == MenuAction.SPELL_CAST_ON_NPC.getId())
+		{
+			client.setMenuEntries(ArrayUtils.removeElement(client.getMenuEntries(), event.getMenuEntry()));
```

Comment:
Rather than doing this set client.menuEntryCount to client.menuEntryCount - 1

Reply:
didn't know you could do this, awesome thanks